internal/charmstore/addentity.go

```diff
@@ -437,7 +437,7 @@ func (s *Store) addCharm(c charm.Charm, p addParams) (err error) {
 	// the name of a bundle. This is racy, but it's the best we can
 	// do. Also check that there isn't an existing multi-series charm
 	// that would be replaced by this one.
-	entities, err := s.FindEntities(entity.BaseURL, nil)
+	entities, err := s.FindEntities(entity.BaseURL, mongodoc.UnpublishedChannel, nil)
```

Comment:
It looks like this is finding just unpublished entities. Isn't that wrong here? For example what if there is a duplicate name but all its entities are published?